x-pack/plugins/security_solution/public/timelines/components/timeline/properties/helpers.tsx

```diff
@@ -104,55 +130,153 @@ interface DescriptionProps {
-          data-test-subj="timeline-description"
-          fullWidth={true}
-          onChange={(e) => updateDescription({ id: timelineId, description: e.target.value })}
-          placeholder={i18n.DESCRIPTION}
-          spellCheck={true}
-          value={description}
-        />
+  ({
+    description,
+    timelineId,
+    updateDescription,
+    isTextArea = false,
+    disableAutoSave = false,
+    disableTooltip = false,
+    disabled = false,
+    marginRight,
+  }) => {
+    const onDescriptionChanged = useCallback(
+      (e) => {
+        updateDescription({ id: timelineId, description: e.target.value, disableAutoSave });
+      },
+      [updateDescription, disableAutoSave, timelineId]
+    );
+
+    const inputField = useMemo(
+      () =>
+        isTextArea ? (
+          <EuiTextArea
+            data-test-subj="timeline-description-textarea"
+            aria-label={i18n.TIMELINE_DESCRIPTION}
+            fullWidth={true}
+            onChange={onDescriptionChanged}
+            placeholder={i18n.DESCRIPTION}
+            value={description}
+            disabled={disabled}
+          />
+        ) : (
+          <EuiFieldText
+            aria-label={i18n.TIMELINE_DESCRIPTION}
+            data-test-subj="timeline-description"
+            fullWidth={true}
+            onChange={onDescriptionChanged}
+            placeholder={i18n.DESCRIPTION}
+            spellCheck={true}
+            value={description}
+          />
+        ),
+      [description, isTextArea, onDescriptionChanged, disabled]
+    );
+    return (
+      <DescriptionContainer data-test-subj="description-container" marginRight={marginRight}>
+        {disableTooltip ? (
+          inputField
+        ) : (
+          <EuiToolTip
+            data-test-subj="timeline-description-tool-tip"
+            content={i18n.DESCRIPTION_TOOL_TIP}
+          >
+            {inputField}
+          </EuiToolTip>
+        )}
       </DescriptionContainer>
-    </EuiToolTip>
-  )
+    );
+  }
 );
 Description.displayName = 'Description';
 
 interface NameProps {
+  autoFocus?: boolean;
+  className?: string;
+  disableAutoSave?: boolean;
+  disableTooltip?: boolean;
+  disabled?: boolean;
   timelineId: string;
   timelineType: TimelineType;
   title: string;
   updateTitle: UpdateTitle;
+  width?: string;
+  marginRight?: number;
 }
 
-export const Name = React.memo<NameProps>(({ timelineId, timelineType, title, updateTitle }) => {
-  const handleChange = useCallback((e) => updateTitle({ id: timelineId, title: e.target.value }), [
+export const Name = React.memo<NameProps>(
+  ({
+    autoFocus = false,
+    className = TIMELINE_TITLE_CLASSNAME,
+    disableAutoSave = false,
+    disableTooltip = false,
+    disabled = false,
     timelineId,
+    timelineType,
+    title,
     updateTitle,
-  ]);
+    width,
+    marginRight,
+  }) => {
+    const handleChange = useCallback(
+      (e) => updateTitle({ id: timelineId, title: e.target.value, disableAutoSave }),
```

Comment:
maybe we should add debounce to don't dispatch `updateTitle` on every key press?

Reply:
It seems that there's already a delay 500ms in x-pack/plugins/security_solution/public/timelines/store/timeline/epic.ts Line 500, so I think it's ok here